src/Sema.zig

```diff
@@ -26486,6 +26487,19 @@ fn zirWorkItem(
     });
 }
 
+fn zirTypeId(
+    sema: *Sema,
+    block: *Block,
+    extended: Zir.Inst.Extended.InstData,
+) CompileError!Air.Inst.Ref {
+    const extra = sema.code.extraData(Zir.Inst.UnNode, extended.operand).data;
+    const ty_src: LazySrcLoc = .{ .node_offset_builtin_call_arg0 = extra.node };
+
+    const ty = try sema.resolveType(block, ty_src, extra.operand);
+
+    return sema.mod.intRef(Type.u32, @intFromEnum(ty.ip_index));
```

Comment:
Is this a comptime value? Will this be stable across compiler runs?

Reply:
Yes, it is a comptime value. I believe it will be stable across compiler runs with this implementation (as long as you're compiling the same source with the same options), but a proposal will have to figure out whether or not the `@typeId`s should even be the same across runs, and if so, how.

Reply:
If its comptime then it will need to be stable across all compiler runs ever, because again we'd have nondeterminism in the language. If its runtime, this can be side stepped and made implementation-defined behavior (because it only needs to be stable across compiler runs & it can change across versions).

Maybe this seems like an arbitrary distinguishment, but I think it's right that returning an undefined value at runtime is fine.